src/metabase/actions/execution.clj

```diff
@@ -199,16 +199,20 @@
              (tru "Values can only be fetched for actions that require a Primary Key."))
   (let [implicit-action (keyword (:kind action))
         {:keys [prefetch-parameters]} (build-implicit-query action implicit-action request-parameters)
-        info {:executed-by api/*current-user-id*
-              :context :question
-              :dashboard-id dashboard-id}
-        card (t2/select-one Card :id (:model_id action))
+        info             {:executed-by api/*current-user-id*
+                          :context :question
+                          :dashboard-id dashboard-id}
+        card             (t2/select-one Card :id (:model_id action))
         ;; prefilling a form with day old data would be bad
-        result (binding [persisted-info/*allow-persisted-substitution* false]
-                 (qp/process-query-and-save-execution!
-                   (qp.card/query-for-card card prefetch-parameters nil nil)
-                   info))
-        exposed-params (set (map :id (:parameters action)))]
+        result           (binding [persisted-info/*allow-persisted-substitution* false]
+                           (qp/process-query-and-save-execution!
+                            (qp.card/query-for-card card prefetch-parameters nil nil)
+                            info))
+        ;; only expose values for fields that are not hidden
+        hidden-param-ids (keep #(when (:hidden %) (:id %))
+                               (vals (get-in action [:visualization_settings :fields])))
+        exposed-params   (-> (set (map :id (:parameters action)))
```

Comment:
```suggestion
        exposed-params-ids (-> (set (map :id (:parameters action)))
```

Reply:
Updated [Rename to exposed-params-ids](https://github.com/metabase/metabase/pull/30933/commits/d86288940e1b23ede7a7b77abfbf5cf4cc399564)